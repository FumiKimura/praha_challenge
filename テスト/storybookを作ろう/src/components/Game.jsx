import { Board } from './Board';

export function Game({ history, currentMove, setCurrentMove, setHistory }) {
  const xIsNext = currentMove % 2 === 0;
  const currentSquares = history[currentMove];

  function handlePlay(nextSquares) {
    const nextHistory = [...history.slice(0, currentMove + 1), nextSquares];
    setHistory(nextHistory);
    setCurrentMove(nextHistory.length - 1);
  }

  function jumpTo(nextMove) {
    setCurrentMove(nextMove);
  }

  const moves = history.map((squares, move) => {
    let description;
    if (move > 0) {
      description = 'Go to move #' + move;
    } else {
      description = 'Go to game start';
    }
    return (
      <li key={move}>
        <button onClick={() => jumpTo(move)}>{description}</button>
      </li>
    );
  });

  return (
    <div className="game">
      <div className="game-board">
        <Board
          xIsNext={xIsNext}
          squares={currentSquares}
          onPlay={handlePlay}
          draw={history.length == 10} // there are 10 total moves (states)
        />
      </div>
      <div className="game-info">
        <ol>{moves}</ol>
      </div>
    </div>
  );
}
