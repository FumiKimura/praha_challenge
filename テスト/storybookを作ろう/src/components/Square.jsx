export function Square({ value, onSquareClick, index }) {
  return (
    <button
      className="square"
      style={{ color: 'ed' }}
      data-e2e={`square-${index}`}
      onClick={onSquareClick}
    >
      {value}
    </button>
  );
}
