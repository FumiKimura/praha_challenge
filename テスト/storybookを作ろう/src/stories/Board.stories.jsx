import { Board } from '../components/Board';

export default {
    component: Board,
    title: 'Board',
    tags: [],
};

export const Default = {
    args: {
        xIsNext: true,
        squares: Array(9).fill(null),
        onPlay: () => {},
    },
};