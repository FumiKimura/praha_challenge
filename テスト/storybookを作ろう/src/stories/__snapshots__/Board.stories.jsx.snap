// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Default smoke-test 1`] = `
<div class="status">
  次のプレーヤー: X
</div>
<div class="board-row">
  <button class="square">
  </button>
  <button class="square">
  </button>
  <button class="square">
  </button>
</div>
<div class="board-row">
  <button class="square">
  </button>
  <button class="square">
  </button>
  <button class="square">
  </button>
</div>
<div class="board-row">
  <button class="square">
  </button>
  <button class="square">
  </button>
  <button class="square">
  </button>
</div>
`;
